pkg/cmd/roachtest/test_runner.go

```diff
@@ -643,6 +648,14 @@ func (r *testRunner) runWorker(
 				}
 				qp.Release(alloc)
 				alloc = nil
+
+				// stop the tests if the failure rate has been exceeded
+				r.status.Lock()
+				failureRate := float64(len(r.status.fail)) / float64(r.testCount)
```

Comment:
I don't have a strong opinion on this yet, so I'm curious about what others think: should all errors be the same when it comes to this (i.e., test failures and infra flakes), or should we not be counting infra flakes (would we want to abort runs if we have extra preemptions on some night for some reason, for example?)

Reply:
I think that's reasonable to not count infra flakes. If the intention of this flag is to prevent excessive triaging and closing of issues, then it makes sense to not count infra flakes that are already rerouted to us and mostly ignored.

Reply:
Agreed. Let's exclude all _known_ infra failures since those are already handled in a specific way to prevent noise.

Reply:
Updated